src/Data/String/Regex.purs

```diff
@@ -121,14 +121,23 @@ foreign import _match
   """
   function _match(r, s, Just, Nothing) {
     var m = s.match(r);
-    return m == null ? Nothing : Just(m);
+    if (m == null) {
+      return Nothing;
+    } else {
+      var list = [];
+      for (var i = 0; i < m.length; i++) {
+        list.push(m[i] == null ? Nothing : Just(m[i]));
+      }
+      return Just(list);
+    }
   }
   """ :: forall r. Fn4 Regex String ([String] -> r) r r
```

Comment:
It think we should fix this type signature now, since we are actually using `Nothing` and `Just` at two different instantiations:

```purescript
Fn4 Regex String (forall r. r -> Maybe r) (forall r. Maybe r) (Maybe (Maybe r))
```